Apps/FlightSoftware/Watchdog2/src/uart.c

```diff
@@ -0,0 +1,393 @@
+/*
+ * Code for interfacing with UART protocol hardware module.
+ *
+ * UART communication uses eUSCI_A0 (Hercules) and eUSCI_A1 (Lander).
+ *
+ * 9600 baud, 1 stop bit, no parity
+ */
+#include <msp430.h>
+#include "include/uart.h"
+#include "include/flags.h"
+
+/* end byte */
+#define SLIP_END 0xC0
+/* escape byte */
+#define SLIP_ESC 0xDB
+/* escaped end */
+#define SLIP_ESC_END 0xDC
+/* escaped escape */
+#define SLIP_ESC_ESC 0xDD
+
+/* ========================== interrupt handlers =========================== */
+
+/* uart0 rx interrupt handler (Hercules) */
+volatile uint8_t uart0_rx_mode = UA0_RX_HEADER;
+volatile uint8_t uart0_rx_header[8] = {0};
+volatile uint16_t uart0_rx_len = 0;
```

Comment:
After moving the uart data parsing logic out of the ISR (which should be done) these should't need to be volatile or global. I'd imagine they could be static variables in whatever new function you use to do your UART RX data processing.